File: python/packages/autogen-core/src/autogen_core/_message_context.py
from dataclasses import dataclass

from ._agent_id import AgentId
from ._cancellation_token import CancellationToken
from ._topic import TopicId


@dataclass
class MessageContext:
    sender: AgentId | None
    topic_id: TopicId | None
    is_rpc: bool
    cancellation_token: CancellationToken
    message_id: str

File: dotnet/src/Microsoft.AutoGen/Contracts/PythonEquiv/MessageContext.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// MessageContext.cs

namespace Microsoft.AutoGen.Contracts.Python;

public class MessageContext(string messageId, CancellationToken cancellationToken)
{
    public string MessageId { get; set; } = messageId;
    public CancellationToken CancellationToken { get; set; } = cancellationToken;

    public AgentId? Sender { get; set; }
    public TopicId? Topic { get; set; }

    public bool IsRpc { get; set; }
}

